### PR TITLE
pod: add new WithLabels method

### PR DESCRIPTION
**Pull request by jmontesi**
The existing `WithLabel` method only allows to add one label to a Pod's definition. If more want to be added the old ones will be overwritten. This new method accepts a set of labels in the form of a `map` to define all the labels required at once.

Also, the `isMutationAllowed` and the `isMountAlreadyInUseInPod` methods haven been modified to return a `bool` as the name suggests. This has been used in the different `With*` methods to leave as early as possible if the right condition is not satisfied.

### CODE REVIEW

**Review by kononovn — CHANGES_REQUESTED** [review]
@jmontesi Thanks for the contribution. Let's change this PR a bit.
Can you please just add single WithLabels function to this PR.

As for isMutationAllowed change we need to create issue and discuss it. If it's going to be accepted we'll need to refactor **all** packages under this project because this design pattern is applied across all packages and not in single pod pkg.

**Review by kononovn — CHANGES_REQUESTED** [review]
Overall looks good. Just 1 comment to fix.

File: pkg/pod/pod.go [hunk 1]
```diff
@@ -898,6 +898,25 @@ func (builder *Builder) WithLabel(labelKey, labelValue string) *Builder {
 	return builder
 }
 
+// WithLabels applies a set of labels to a Pod's definition.
+func (builder *Builder) WithLabels(labels map[string]string) *Builder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
```
[the remark this answers is not in the document]
Comment (jmontesi) [reply pkg/pod/pod.go:906]: Done.

**Review by kononovn — DISMISSED** [review]
/lgtm

**Review by kononovn — APPROVED** [review]

**Pull request merged**